run.php

```diff
@@ -80,6 +80,12 @@
     InputOption::VALUE_NONE,
     'Do not delete _data and _modules directories before running'
 ];
+$optionUnsupported = [
+    'unsupported',
+    null,
+    InputOption::VALUE_NONE,
+    'Only update unsupported modules that were supported in the previous CMS major. Must also use --branch=default-branch. Can not be used with --cms-major option.'
```

Comment:
Do you mean `--branch=github-default`? Or is `--branch=default-branch` a  new option?

I think using `--branch` at all should be considered an error, and `--unsupported` should set to the default branch without having to explicitly pass that in.

What's more, I don't see why it can't be used with `--cms-major` option - we might want to run this explicitly against CMS 6 before that goes stable, for example.

Reply:
Sorry yes should be `--branch=github-default`

> I think using --branch at all should be considered an error, and --unsupported should set to the default branch without having to explicitly pass that in.

I considered that, however I opted again it because it makes it less clear from the command line what's going on. Having explicitly specify "Doing this on the default branch" makes it more obvious what's going on, because the `--unsupported` option by itself to me intuitively means "Do things on the CMS 4 branch" which is not the case at all.

> What's more, I don't see why it can't be used with --cms-major option - we might want to run this explicitly against CMS 6 before that goes stable

I don't understand this what's intended here? `--unsupported` literally means do things on modules that no longer supported.  Unsupported is determined from looking at silverstripe/supported-modules repositories.json.  It makes no sense "running this on CMS 6" because everything in CMS 5 is still supported

Reply:
I've agreed to ignore most of the things that bug me about this in the interest of moving forward - but as discussed in person we really shouldn't need to have the two flags. Please rename `--unsupported` to something that describes what it does in a way that doesn't require having two flags for the sake of clarity.